src/parser.rs

```diff
@@ -187,6 +187,38 @@ pub fn parse_instruction(input: &[u8]) -> IResult<&[u8], Opcode, VerboseError<&[
         0x15 => (i, Opcode::Dec8(Register8::D)),
         0x25 => (i, Opcode::Dec8(Register8::H)),
         0x35 => (i, Opcode::Dec8(Register8::DerefHL)),
+
+        // the entire 40..7F block 
+        0x76 => (i, Opcode::Halt), // (0x76 case must come before the 0x40..=0x7F case)
+        0x40..=0x7F => {
```

Comment:
⛳️ `0x40..0x76 | 0x77..=0x7F` 

then we don't need the comment above, makes the code simpler if read non-linearly

Reply:
Github was weird so I wrote this comment twice 🤷‍♂